common/factorization/par_ilut_select_kernels.hpp.inc

```diff
@@ -62,7 +62,8 @@ __global__ __launch_bounds__(searchtree_width) void build_searchtree(
     // assuming rounding towards zero
     auto stride = double(size) / sample_size;
 #pragma unroll
-    for (auto i = 0; i < sampleselect_oversampling; ++i) {
+    for (auto i = decltype(sampleselect_oversampling){0};
+         i < sampleselect_oversampling; ++i) {
```

Comment:
that completely destroys readability IMO.
```suggestion
    for (int i = 0; i < sampleselect_oversampling; ++i) {
```

Reply:
I agree, directly stating the type is better IMO, even if it makes it slightly less adaptable (to type changes).